test/storage/export_table_test.cpp

```diff
@@ -0,0 +1,312 @@
+#include <unordered_map>
+#include <vector>
+
+#include "common/hash_util.h"
+#include "storage/block_access_controller.h"
+#include "storage/block_compactor.h"
+#include "storage/garbage_collector.h"
+#include "storage/storage_defs.h"
+#include "storage/tuple_access_strategy.h"
+#include "test_util/storage_test_util.h"
+#include "test_util/test_harness.h"
+#include "transaction/deferred_action_manager.h"
+
+#define EXPORT_TABLE_NAME "test_table.arrow"
+#define CSV_TABLE_NAME "test_table.csv"
+#define PYSCRIPT_NAME "transform_table.py"
+#define PYSCRIPT                                      \
+  "import pyarrow as pa\n"                            \
+  "pa_table = pa.ipc.open_stream('" EXPORT_TABLE_NAME \
+  "').read_next_batch()\n"                            \
+  "pa_table = pa_table.to_pandas()\n"                 \
+  "pa_table.to_csv('" CSV_TABLE_NAME "', index=False, header=False)\n"
+
+namespace terrier {
+
+struct ExportTableTest : public ::terrier::TerrierTest {
+  bool ParseNext(std::ifstream &csv_file, char stop_char, char *tmp_char) {
+    csv_file.get(*tmp_char);
+    bool end = false;
+    switch (*tmp_char) {
```

Comment:
Could you comment the differences between these cases and the logic to handle them?